### PR TITLE
Extend validations

### PR DESCRIPTION
**Pull request by gruz0**
1. Updated validation errors descriptions
2. Disabled registration without login
3. Fixed bug when budget is not a number in create/update job
4. Removed extra whitespaces from email and display_name on registration
5. Added validations if login or password are empty on registration
6. Added automatically generated `display_name` on registration if it is empty

### CODE REVIEW

File: pkg/db/pgdao/contracts.sql.go [hunk 1]
```diff
@@ -13,24 +13,27 @@ import (
 
 const contractAdd = `-- name: ContractAdd :one
 insert into contracts (
-    id, customer_id, performer_id, application_id, title, description, price, duration, created_by, customer_address
+    id, customer_id, performer_id, application_id, title, description, price, duration, created_by, customer_address, performer_address, status, contract_address
```
Comment (almaz-uno) [inline pkg/db/pgdao/contracts.sql.go:16]: it's reasonable to switch named parameters here for readability sake

File: pkg/service/pgsvc/job.go [hunk 2]
```diff
@@ -31,7 +31,7 @@ func (s *JobSvc) Add(ctx context.Context, e *model.Job) (*model.Job, error) {
 		if e.Budget.IsNegative() {
 			return &model.BackendError{
 				Cause:   model.ErrValidationFailed,
-				Message: model.ValidationErrorMustBePositive("budget"),
+				Message: model.ValidationErrorMustBePositive("Budget"),
```
Comment (almaz-uno) [inline pkg/service/pgsvc/job.go:34]: it would still be cool to title budget → Budget on the frontend

File: intest/job_test.go [hunk 3]
```diff
@@ -50,6 +68,52 @@ func TestJob(t *testing.T) {
 		}
 	})
 
+	t.Run("post•without wallet", func(t *testing.T) {
+		body := `{
+			"title":"Create awesome site",
+			"description": "There are words here. Very much words."
```
Comment (almaz-uno) [inline intest/job_test.go]: very many words

File: pkg/controller/jobs.go [hunk 4]
```diff
@@ -64,23 +65,39 @@ func (cont *Job) add(c echo.Context) error {
...
+		return &model.BackendError{
+			Cause:   model.ErrValidationFailed,
+			Message: "Ethereum address is required",
+		}
+	}
+
 	ie := new(jobDescription)
 
 	if e := c.Bind(ie); e != nil {
+		re := regexp.MustCompile("error decoding string.*to decimal")
```
Comment (almaz-uno) [inline pkg/controller/jobs.go]: it will be implemented  in handlers.go

File: pkg/controller/jobs.go [hunk 5]
```diff
@@ -167,6 +184,15 @@ func (cont *Job) update(c echo.Context) error {
 	id := c.Param("id")
 
 	if e := c.Bind(&ie); e != nil {
+		re := regexp.MustCompile("error decoding string.*to decimal")
```
Comment (almaz-uno) [inline pkg/controller/jobs.go]: it will be implemented in handlers.go

File: pkg/db/pgdao/applications.sql.go [hunk 6]
```diff
@@ -51,20 +51,25 @@ func (q *Queries) ApplicationAdd(ctx context.Context, arg ApplicationAddParams)
 
 const applicationGet = `-- name: ApplicationGet :one
 
-select a.id, a.created_at, a.updated_at, a.comment, a.price, a.job_id, a.applicant_id, c.id as contract_id from applications a
-	left join contracts c on a.id = c.application_id
+select a.id, a.created_at, a.updated_at, a.comment, a.price, a.job_id, a.applicant_id,
+	(CASE WHEN p.display_name = '' THEN p.login ELSE p.display_name END)::varchar AS applicant_display_name,
```
Comment (almaz-uno) [inline pkg/db/pgdao/applications.sql.go:55]: A little step to procedural language in PostgreSQL

File: pkg/model/errors.go [hunk 7]
```diff
@@ -38,12 +38,13 @@ var (
 	ErrValidationFailed         = errors.New("validation failed")
 	ErrInvalidFormat            = errors.New("invalid format")     // entire request body has invalid format
 	ErrInsufficientFunds        = errors.New("insufficient funds") // insufficient funds on some address in a blockchain network
+	ErrInvalidJSON              = errors.New("invalid JSON")
```
Comment (almaz-uno) [inline pkg/model/errors.go:41]: Is this error type still used in the code?

File: pkg/service/pgsvc/contract.go [hunk 8]
```diff
@@ -93,238 +178,258 @@ func (s *ContractSvc) Add(ctx context.Context, contract *model.Contract) (*model
...
 
-		if performerAddress == c.CustomerAddress {
+		if !common.IsHexAddress(c.ContractAddress) {
 			return &model.BackendError{
-				Cause:    model.ErrInsufficientFunds,
-				Message:  "customer and performer addresses cannot be the same",
-				TechInfo: "performer_address",
+				Cause:    model.ErrValidationFailed,
+				Message:  model.ValidationErrorInvalidFormat("contract_address"),
+				TechInfo: c.ContractAddress,
```
Comment (almaz-uno) [inline pkg/service/pgsvc/contract.go:260]: Should be here name of the field, not value?

File: intest/job_test.go (hunk 3 above)
Comment (gruz0) [reply intest/job_test.go]: Fixed

File: pkg/model/errors.go (hunk 7 above)
Comment (gruz0) [reply pkg/model/errors.go:41]: It is:

```
[13:03] optriment/optrispace-backend ‹refactor/extend-validations*›$ grep -r ErrInvalidJSON .
./pkg/web/handlers.go:                          cause = model.ErrInvalidJSON // "invalid JSON"
./pkg/model/errors.go:  ErrInvalidJSON              = errors.New("invalid JSON")
```

File: pkg/service/pgsvc/contract.go (hunk 8 above)
Comment (gruz0) [reply pkg/service/pgsvc/contract.go:260]: It would be good to talk about it later

**Pull request merged**